### PR TITLE
Converted everything to TypeScript

### LINKED ISSUE #26: Refactor tests to typescript
The main source code has been migrated to typescript. The tests' code should also be migrated.

WIP on branch: `tests-typescript`

### PR DESCRIPTION
**Pull request by radekmie**
In this PR, I've included #28 and converted everything else to TypeScript. All tests are passing and everything works as expected. Some of the types are incorrect, but I'd rather focus on that in a following PR.

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -3,10 +3,12 @@
     "baseUrl": "packages",
     "composite": true,
     "declaration": true,
+    "esModuleInterop": true,
     "importHelpers": true,
-    "lib": ["es2017", "esnext.asynciterable"],
-    "module": "commonjs",
+    "module": "CommonJS",
+    "moduleResolution": "Node",
+    "paths": { "sparrowql": ["sparrowql/src"], "sparrowql-*": ["sparrowql-*/src"] },
```
Comment (wadamek65) [inline tsconfig.json:10]: Would just `"sparrowql*": ["sparrowql*/src"]` suffice?

File: __tests__/demo.ts [hunk 2]
```diff
@@ -0,0 +1,155 @@
+import { build } from 'sparrowql';
+import { v1 as uuid } from 'uuid';
+
+testWithNCollections(
+  'demo',
+  [uuid(), uuid(), uuid()] as const,
+  async (Blogs, Posts, Users) => {
+    const blogs = [
+      { _id: 0, ownerId: 3, topic: 'Best blog!' }, // prettier ignore
```
Comment (wadamek65) [inline __tests__/demo.ts]: Is this an actual working directive?

**Review by wadamek65 — APPROVED** [review]
I only generally checked the new TS files and didn't try to compare the changes to the previous ones.

File: tsconfig.json (hunk 1 above)
Comment (radekmie) [reply tsconfig.json:10]: Nope, it's not working - `sparrowql` is not accessible then.

**Review by wadamek65 — APPROVED** [review]

**Review by Monteth — APPROVED** [review]

**Review by zendranm — APPROVED** [review]

**Pull request merged**